tensorflow/core/util/tensor_bundle/byte_swap_tensor.cc

```diff
@@ -109,6 +109,43 @@ Status ByteSwapBuffer(char* buff, size_t size, DataType dtype,
 
 }  // namespace
 
+bool IsByteSwappable(DataType dtype) {
+  switch (dtype) {
+    // 16-bit types
+    case DT_BFLOAT16:
+    case DT_HALF:
+    case DT_QINT16:
+    case DT_QUINT16:
+    case DT_UINT16:
+    case DT_INT16:
+
+    // 32-bit types
+    case DT_FLOAT:
+    case DT_INT32:
+    case DT_QINT32:
+    case DT_UINT32:
+
+    // 64-bit types
+    case DT_INT64:
+    case DT_DOUBLE:
+    case DT_UINT64:
+
+    // Complex types
+    case DT_COMPLEX64:
+    case DT_COMPLEX128:
+      return true;
+    
+    // Types that ought to be supported in the future
```

Comment:
Are you sure these ought to be byte-swappable?  They are generally not POD tensors.

Reply:
Hi @cantonios ,

Yes, these data types should be byte-swappable. They're already used in the existing `ByteSwapBuffer()` function.

Reply:
We do byte swapping for these data types when they're used in `tensor_content` field of `Const op` between different endianness platforms, because these data are stored in host endianness format. I am not very sure what do you mean by POD tensors and their relation to endianness conversion. Could you please explain more on this?

Thank you very much! @cantonios

Reply:
POD = Plain-old-data

A Resource can have a pointer to some internal data resource stored in its tensor content.  And could have arbitrary data, not necessarily all of a fixed type, so you likely will not be able to simply swap bytes around.  Similarly, AFAIK Variant could be anything, so I'm not sure how you will know how to byte-swap it, since you don't know the size of elements to swap around.

Reply:
Hi @cantonios ,

Sorry I misunderstood you just now. I thought you were referring to the byte-swappable data types such as `DT_INT32`.

Yes, you're right, we may not be able to do byte swapping for `DT_RESOURCE` and `DT_VARIANT`.  I'll remove relevant lines from `IsByteSwappable()`.

Reply:
I've removed `DT_RESOURCE` and `DT_VARIANT` related lines from `IsByteSwappable()`. Please take a look when you have some time. Thanks @cantonios !